waku/v2/protocol/relay/waku_relay.go

```diff
@@ -227,9 +227,38 @@ func (w *WakuRelay) Subscribe(ctx context.Context, topic *Topic) (*Subscription,
 	return subscription, nil
 }
 
+func (w *WakuRelay) Unsubscribe(ctx context.Context, topic *Topic) error {
+	if _, ok := w.topics[*topic]; !ok {
+		return fmt.Errorf("topics %s is not subscribed", (string)(*topic))
+	}
+	log.Info("Unsubscribing from topic ", *topic)
+	delete(w.topics, *topic)
+
+	for _, sub := range w.subscriptions[*topic] {
+		sub.Unsubscribe()
+	}
+
+	w.relaySubs[*topic].Cancel()
+	delete(w.relaySubs, *topic)
+
+	err := w.wakuRelayTopics[*topic].Close()
+	if err != nil {
+		return err
+	}
+	delete(w.wakuRelayTopics, *topic)
+
+	return nil
+}
+
 func (w *WakuRelay) nextMessage(ctx context.Context, sub *pubsub.Subscription) <-chan *pubsub.Message {
 	msgChannel := make(chan *pubsub.Message, 1024)
 	go func(msgChannel chan *pubsub.Message) {
+		defer func() {
+			if r := recover(); r != nil {
```

Comment:
TIL about `recover()`!

Reply:
Yeah I am not sure if this is the best way but `msgChannel <- msg` cause a panic, I think to improve it we could intercept the close/quit and handle that properly